### ISSUE TITLE
error: Unable to load the service index for source https://MyAccount.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/MyProject/nuget/index.json. 

### ISSUE THREAD

**Issue by TheRealPiotrP** [NuGet/Home#3929]
Moving https://github.com/dotnet/cli/issues/4624 on behalf of @kievic

----

## Steps to reproduce

1. Open Visual Studio 2015
2. Go to Tools > Options ....
3. Navigate to NuGet Package Manager > Package Sources
4. Click + button (add package source)
5. Write a name in the Name field
6. Set the Source field to a NuGet source that requires Azure Active Directory authentication, such as, `https://contsoso.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/ContosoNuGet/nuget/index.json`
7. Then do a `dotnet restore .\project.json` on any project.

## Expected  behavior

This should complete with a successful message:

> log  : Restoring packages for D:\repos\Foo\project.json
> ...
> log  : Restore completed in 3679ms.

## Actual behavior

The `dotnet` client fails with a very ambiguous error that could take hours to understand. There is no clue that the failing package source was set on Visual Studio, and it is not referenced by the project we are restoring.

> log  : Restoring packages for D:\repos\Foo\project.json
> error: Unable to load the service index for source https://consotoso.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/ContosoNuGet/nuget/index.json.
> error:   An error occurred while sending the request.
> error:   The parameter is incorrect

Moreover, NuGet package sources with Azure Active Directory authentication work well with traditional desktop solutions when restoring packages from Visual Studio.

## Environment data

```
.NET Command Line Tools (1.0.0-preview2-003131)

Product Information:
 Version:            1.0.0-preview2-003131
 Commit SHA-1 hash:  635cf40e58

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by SamHard** [comment]
In the case of a private NuGet feed using Azure Active Directory authentication, the work around is to manually install to your local cache each package version that you want to use, using the Package Manager Console's Install-Package command.  Here's a sample:

`Install-Package Contoso.Binary.DataAccess -Source "https://contsoso.pkgs.visualstudio.com/DefaultCollection/_apis/packaging/ContosoNuGet/nuget/index.json" -Version 0.1.1`

You only have to do this the first time you want to reference a particular package version.

**Comment by anupBd** [comment]
I have used generator-aspnetcore-spa to generate a project. Then when I tried to restore project using dotnet restore command I got the same error. I figured out that restore use "C:\Users\**\AppData\Roaming\NuGet\NuGet.Config" file for the config of package restore and there I got activePackageSource was a private feed that I am using for my other project which I have configured from visualstudio. Then I open the Manage nuget package for the solution and then choose nuget.org as feed, then the NuGet.Config file looks like bellow and then dotnet restore was successful
`<packageSources>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" protocolVersion="3" />
  </packageSources>
  <activePackageSource>
    <add key="nuget.org" value="https://api.nuget.org/v3/index.json" />
  </activePackageSource>`
